src/main/java/org/springframework/data/elasticsearch/client/elc/IndicesTemplate.java

```diff
@@ -241,6 +248,45 @@ public Map<String, Object> getMapping() {
 		return responseConverter.indicesGetMapping(getMappingResponse, indexCoordinates);
 	}
 
+	@Override
+	public ClusterMapping getClusterMapping() {
+		GetMappingRequest getMappingRequest = requestConverter.indicesGetMappingRequest(IndexCoordinates.of("*"));
+		GetMappingResponse getMappingResponse = execute(client -> client.getMapping(getMappingRequest));
+
+		return responseConverter.indicesGetMapping(getMappingResponse);
+	}
+
+	/**
+	 * Refreshes the mapping of entities.
+	 * <p>
+	 * This method is responsible for retrieving and updating the metadata related to the entities.
+	 */
+	private void refreshEntitiesMapping() {
+		ClusterMapping clusterMapping = getClusterMapping();
+		for (ClusterMapping.ClusterMappingEntry mappingEntry : clusterMapping) {
+			// Get entity
+			Class<?> entity = null;
+			for (ElasticsearchPersistentEntity<?> persistentEntity : this.elasticsearchConverter.getMappingContext().getPersistentEntities()) {
+				if (mappingEntry.getName().equals(persistentEntity.getIndexCoordinates().getIndexName())) {
+					entity = persistentEntity.getType();
+
+					break;
+				}
+			}
+
+			if (entity == null) {
+				continue;
+			}
+
+			if (mappingEntry.getMappings().containsKey("dynamic_templates")) {
+				Object dynamicTemplates = mappingEntry.getMappings().get("dynamic_templates");
+				if (dynamicTemplates instanceof List<?> value) {
+					getRequiredPersistentEntity(entity).buildDynamicTemplates(value);
```

Comment:
in line 269 we have that persistentEntity, so there would be no need to call `getRequiredPersistentEntity`